cmd/kubeadm/app/cmd/init.go

```diff
@@ -397,6 +400,9 @@ func (d initData) KubeConfigDir() string {
 
 // KubeConfigPath returns the path to the kubeconfig file to use for connecting to Kubernetes
 func (d initData) KubeConfigPath() string {
+	if d.dryRun {
+		d.kubeconfigPath = filepath.Join(d.dryRunDir, kubeadmconstants.AdminKubeConfigFileName)
```

Comment:
that was a bug indeed, thanks for fixing. @yuexiao-wang

Reply:
thanks for your help